### PR TITLE
Patch bash to load startup files when invoked with '-bash'

### PR DESCRIPTION
**Pull request by dmcilvaney**
i.e., when invoked via 'su - <user> -c ...'. This is done by Fedora and Ubuntu (and other distros).

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [x] Ready to merge

---

###### Summary 

Bash loads startup scripts when it runs, based on some logic to detect if it is a login/normal prompt, and interactive/non-interactive. (profile, bashrc, etc..., depending on those modes).

Tools like `su` in `util-linux` can explicitly request login/non-login mode (`su - other_user ...` (`-` == `-l` == `--login`). It informs bash of this choice by  prepending a '-' to `argv[0]` (i.e., `-bash`). By default `bash` ignores this, but can be [configured to respect it via a "hidden" define](https://git.savannah.gnu.org/cgit/bash.git/tree/config-top.h#n105) `NON_INTERACTIVE_LOGIN_SHELLS`. Most distros set this flag, so lets add it.

In containers:
```bash
tdnf -y install util-linux shadow-utils # 'dnf -y install util-linux' on fedora, ubuntu works out of box
useradd -m test
echo "export test=test" >> /etc/profile
su test -c 'echo $test' # Expected to be ""
su - test -c 'echo $test' # SHOULD BE "TEST", this works on ubuntu/fedora!
```
image

###### Change Log  

- Enable `NON_INTERACTIVE_LOGIN_SHELLS` for bash package.

###### Does this affect the toolchain?  

**YES**

###### Associated issues  

- https://microsoft.visualstudio.com/DefaultCollection/OS/_workitems/edit/46864804

###### Test Methodology

- Local build/test in vm & container
- Buddybuild: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=588741&view=results (uninstall test fails since removing bash from the env. breaks the test).

### CODE REVIEW

File: SPECS/bash/bash.spec [hunk 1]
```diff
@@ -328,6 +330,10 @@ fi
 %defattr(-,root,root)
 
 %changelog
+* Mon Jun 17 2024 Daniel McIlvaney <damcilva@microsoft.com> - 5.2.15-2
+- When non-interactive shells are started with '-bash' load startup files. From
+  Fedora upstream: https://src.fedoraproject.org/rpms/bash/blob/f40/f/bash-2.03-profile.patch
```
Comment (anphel31) [inline SPECS/bash/bash.spec:335]: can you add a dash to the start of this comment?

**Review by anphel31 — APPROVED** [review]
I re-ran the buddybuild with the option set to not remove bash, to confirm it's passing - https://dev.azure.com/mariner-org/mariner/_build/results?buildId=589017&view=results

File: SPECS/bash/bash.spec (hunk 1 above)
Comment (dmcilvaney) [reply SPECS/bash/bash.spec:335]: This is auto-formatted via `rpmdev-bumpspec`, but I can add.

File: SPECS/bash/bash.spec [hunk 2]
```diff
@@ -328,6 +330,10 @@ fi
...
+
```
Comment (dmcilvaney) [inline SPECS/bash/bash.spec:336]:
```suggestion
* Mon Jun 17 2024 Daniel McIlvaney <damcilva@microsoft.com> - 5.2.15-2
- When non-interactive shells are started with '-bash' load startup files. From
- Fedora upstream: https://src.fedoraproject.org/rpms/bash/blob/f40/f/bash-2.03-profile.patch

```

**Pull request merged**